### ISSUE TITLE
Support authentication on .NET Framework application

### ISSUE THREAD

**Issue by Krutonium** [Odonno/gitter-api-auth#4]
This package requires .Net below 4.5, but the other package requires 4.5 to function, as far as I can tell - Am I doing something horrendously wrong?

**Comment by Odonno** [comment]
Indeed, I am unable to install gitter-api-pcl nuget package on a Windows 10 Application.

What type of project are you working on ? I suppose it's a target compatibility issue.

**Comment by Odonno** [comment]
Ok, I found the issue. On a W10 App, you need to install first Microsoft.NET.Http package. The dependency is set for W8.1, WP8.1 but not for W10. I am sorry. I'll fix it asap.

**Comment by Krutonium** [comment]
Thank you!

**Comment by Odonno** [comment]
Hi @PFCKrutonium 

I may have misunderstood your problem. Are you trying to install gitter-api-auth inside a .NET framework application (like WinForms or WPF) ? If that's the case, it is normal that you can't install the package because the only platforms supported by this package are Windows 8.1 and Windows Phone 8.1 store apps (Windows 10 store apps also supported).

**Comment by Krutonium** [comment]
I'm actually trying to use it in a C# console application. How do I set all this up if I cant use this?

**Comment by Odonno** [comment]
Well, it depends on what you want. If you want to try the API for yourself, you can request an authorization token from https://developer.gitter.im/docs/

If you want any user to use your console application, you have to create an OAuth login process. I don't know if it's possible because to do that you have to open a webview. If that's what you need, I will try to find a solution.

**Comment by Odonno** [comment]
@PFCKrutonium That should be done. Can you try the 0.1.4 package and tell me if that works for you ?

The package now works for .NET Framework 4.5 and later. Moreover, you can look at the source code and see that a new sample project has been created, it shows you how to use the package with a C# Console Application.

**Comment by Odonno** [comment]
@PFCKrutonium Any news ?

**Comment by Krutonium** [comment]
Sorry, got pulled away for other things - I'll give it a shot in a couple of hours! Will update with news.

**Comment by Krutonium** [comment]
I tried to copy your example, but I ended up with this:
![screenshot 80](https://cloud.githubusercontent.com/assets/3945538/20551668/b4505ac4-b110-11e6-93c1-5ad7d1c650fe.png)
(After I logged in)

Any ideas?

**Comment by Krutonium** [comment]
Well I fixed that (New OAuth stuff with the Redirect URI set to http://localhost), but it never returns the token. In fact, it just does this. (It does ask me if I want to allow access, of course.)
![screenshot 81](https://cloud.githubusercontent.com/assets/3945538/20552032/afb72750-b114-11e6-98fc-b3fbd9cc19c3.png)

**Comment by Odonno** [comment]
@PFCKrutonium Oh, that's not good. I tried it many times and everything works fine. The redirect URI should be http://localhost exactly.

I'll try again asap to see if I can reproduce the bug.
The default behavior would be not to show you the localhost page but to return to the app with the retrieved token.

**Comment by Odonno** [comment]
I saw that I do not ended the Browser window the first moment we receive the event to go to the localhost page but I got the token as it should be so I am not sure it's the same problem.

Can you try the latest version of the package and see if there is any change ?
